sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/BasicKinesisProvider.java

```diff
@@ -39,16 +40,27 @@
   private final String secretKey;
   private final Regions region;
   private final @Nullable String serviceEndpoint;
+  private final boolean verifyCertificate;
 
   BasicKinesisProvider(
-      String accessKey, String secretKey, Regions region, @Nullable String serviceEndpoint) {
+      String accessKey,
+      String secretKey,
+      Regions region,
+      @Nullable String serviceEndpoint,
+      boolean verifyCertificate) {
     checkArgument(accessKey != null, "accessKey can not be null");
     checkArgument(secretKey != null, "secretKey can not be null");
     checkArgument(region != null, "region can not be null");
     this.accessKey = accessKey;
     this.secretKey = secretKey;
     this.region = region;
     this.serviceEndpoint = serviceEndpoint;
+    this.verifyCertificate = verifyCertificate;
+  }
+
+  BasicKinesisProvider(
```

Comment:
Service endpoint can be configured in the KinesisIO and I thought that omitting it in the producer configuration was a bug.
Certificate verification is needed for cross-language tests (PR [12297](https://github.com/apache/beam/pull/12297)), we would need to use this TestKinesisProvider if verify_certificate was provided there.

Reply:
I'm not sure that service endpoint is used in `com.amazonaws.services.kinesis.producer.KinesisProducer` which is actually used for `KinesisIO.Write`. Am I mistaken?

Reply:
When you use KinesisIO.write()..withAWSClientsProvider("accesskey", "secretkey", "region", "serviceEndpointUrl") then without my modification beam will try to use real AWS url instead of the given "serviceEndpointUrl". I don't think this is the desired behaviour, but correct me if I'm wrong or I don't understand something (which is very probable, I haven't used AWS much in my life).